src/util/util.c

```diff
@@ -181,6 +181,18 @@ int strcicmp(char const *a, char const *b)
 	}
 }
 
+char *safe_strncpy(char * destination, const char * source, size_t num)
+{
+	char *result = strncpy(destination, source, num);
+	int len = strnlen(source, num);
```

Comment:
To do both strncpy and strnlen is inefficient - I don't the compiler can optimise this out either. So might be better to integrate our own loop.